README.md

```diff
@@ -372,28 +375,28 @@ Notes:
-|                  Crystal | 0.122<sub>±0.000</sub> |    2.98<sub>±00.03</sub> + 90.71<sub>±00.17</sub> |   5.52<sub>±00.08</sub> |
-|                     Rust | 0.140<sub>±0.000</sub> |    0.94<sub>±00.01</sub> + 74.26<sub>±00.07</sub> |   5.35<sub>±00.05</sub> |
-|                     Java | 0.159<sub>±0.002</sub> |  38.27<sub>±00.10</sub> + 153.79<sub>±03.78</sub> |   8.89<sub>±00.06</sub> |
-|                  C++/g++ | 0.189<sub>±0.000</sub> |   2.70<sub>±00.90</sub> + 116.66<sub>±00.81</sub> |   7.76<sub>±00.05</sub> |
-|              C++/clang++ | 0.198<sub>±0.000</sub> |    1.66<sub>±00.02</sub> + 87.96<sub>±00.02</sub> |   7.79<sub>±00.09</sub> |
-|                  V/clang | 0.212<sub>±0.001</sub> |   1.91<sub>±00.05</sub> + 203.15<sub>±01.44</sub> |   8.46<sub>±00.06</sub> |
-|                  Node.js | 0.227<sub>±0.002</sub> |  39.10<sub>±00.03</sub> + 150.29<sub>±00.26</sub> |  11.37<sub>±00.12</sub> |
-|                    V/gcc | 0.240<sub>±0.001</sub> |   2.28<sub>±00.14</sub> + 219.14<sub>±00.67</sub> |   9.66<sub>±00.12</sub> |
-|                Nim/clang | 0.297<sub>±0.001</sub> |   2.07<sub>±00.01</sub> + 601.73<sub>±02.96</sub> |  11.72<sub>±00.09</sub> |
-|               Lua/luajit | 0.338<sub>±0.002</sub> |   1.20<sub>±00.02</sub> + 157.12<sub>±01.04</sub> |  13.22<sub>±00.05</sub> |
-|                  Nim/gcc | 0.352<sub>±0.003</sub> |   1.68<sub>±00.07</sub> + 614.62<sub>±00.26</sub> |  13.48<sub>±00.20</sub> |
-|                    Scala | 0.362<sub>±0.006</sub> |  67.43<sub>±00.09</sub> + 248.88<sub>±11.36</sub> |  18.63<sub>±00.35</sub> |
-|                    Julia | 0.597<sub>±0.001</sub> | 246.42<sub>±00.24</sub> + 374.41<sub>±01.06</sub> |  23.23<sub>±00.15</sub> |
-|              Python/pypy | 0.880<sub>±0.002</sub> |  63.32<sub>±00.09</sub> + 250.27<sub>±00.06</sub> |  34.53<sub>±00.42</sub> |
-| Ruby/truffleruby (--jvm) | 1.381<sub>±0.026</sub> | 354.60<sub>±10.97</sub> + 552.99<sub>±34.23</sub> |  90.57<sub>±01.36</sub> |
-|             Ruby (--jit) | 1.442<sub>±0.004</sub> | 270.43<sub>±00.06</sub> + 147.02<sub>±00.05</sub> |  59.14<sub>±00.99</sub> |
-|                      Lua | 1.478<sub>±0.004</sub> |   2.28<sub>±00.01</sub> + 283.04<sub>±00.48</sub> |  57.46<sub>±00.33</sub> |
-|         Ruby/truffleruby | 1.506<sub>±0.007</sub> | 294.80<sub>±00.82</sub> + 410.90<sub>±26.05</sub> |  84.20<sub>±00.88</sub> |
-|               Ruby/jruby | 1.990<sub>±0.057</sub> | 182.28<sub>±01.67</sub> + 538.37<sub>±28.70</sub> | 102.14<sub>±03.48</sub> |
-|                     Ruby | 2.057<sub>±0.005</sub> |  13.87<sub>±00.07</sub> + 147.07<sub>±00.04</sub> |  82.80<sub>±00.47</sub> |
-|                   Python | 5.022<sub>±0.024</sub> |  10.49<sub>±00.04</sub> + 234.72<sub>±00.26</sub> | 196.62<sub>±02.52</sub> |
+|                  C++/g++ | 0.071<sub>±0.000</sub> |    3.68<sub>±00.04</sub> + 77.91<sub>±00.26</sub> |   2.71<sub>±00.02</sub> |
```

Comment:
wonder why c++ became much faster

Reply:
C++ implementation has changed to use the arena allocator (#427). I don't consider it as a micro-optimization as Zig uses the similar allocator too, and it's a totally valid and proper use-case for the embedded systems (where C++ is used a lot).

It's definitely controversial, but we can't have the perfect similarity anyway, and I don't oppose to use the strong sides of the programming languages.

Reply:
Hm, `V` has `-prealloc` argument which turns on arena allocation. Other languages have such flags perhaps too. Shouldn't that be considered here if C++ and Zig have already gotten this "privilege"?

Ping @spytheman .

Reply:
Thanks for the suggestion, it's definitely a useful feature. And while other languages may have it too, I'm not aware of it (including V until you mentioned it, as `v help build` doesn't describe this option). I've taken a look into V source code, apparently it preallocates in 16 Mb chunks that is not that suitable for the embedded systems, however in the future it could be tuned I guess. Nonetheless, I'm fine with using this option, and will provide up-to-dated results soon.

Reply:
Addressed in #444.

Reply:
Hm, I meant only in this benchmark. Or do Zig and C++ use arena allocator for everything?

Reply:
It depends, there are some inconsistencies, and I don't have much time to address everything. But in reality it doesn't matter much - as you could see in the aforementioned PR, only Primes test is really affected.